crates/notion-core/src/inventory/serial.rs

```diff
@@ -74,35 +74,85 @@ impl YarnCollection {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-pub struct Index(Vec<Entry>);
+pub struct NodeIndex(Vec<NodeEntry>);
 
 #[derive(Serialize, Deserialize, Debug)]
-pub struct Entry {
+pub struct NodeEntry {
     pub version: String,
     pub npm: Option<String>,
     pub files: Vec<String>,
 }
 
-impl Index {
-    pub fn into_index(self) -> Fallible<super::Index> {
+fn trim_version(s: &str) -> &str {
+    let s = s.trim();
+    if s.starts_with('v') {
+        s[1..].trim()
+    } else {
+        s
+    }
+}
+
+impl NodeIndex {
+    pub fn into_index(self) -> Fallible<super::NodeIndex> {
```

Comment:
And having `NodeIndex::into_index` (and `YarnIndex::into_index`) is also a little confusing, since it is already an index.